src/matlab2tikz.m

```diff
@@ -2156,7 +2162,10 @@ case size(cData,2) % specific y-data is given
             error('matlab2tikz:UnknownAlphaMapping', ...
                   'Unknown alpha mapping "%s"', alphaMapping);
     end
-    alpha = min(1,max(alpha,0)); % clip at range [0, 1]
+    
+    if isfloat(alpha) %important, alpha data can have integer type which should not be scaled
```

Comment:
Good point, the `hasAlpha` check on L2026 indeed only works correctly for floating point `alphaData`. I've modified L2026 to only check for opaqueness when `alphaData` is float. Always printing the `alphaData` when it is integer is save and should only be inefficient in very rare cases.